python_examples/rebound.py

```diff
@@ -127,4 +149,158 @@ def step():
+    
+    # first check for valid inputs and give warnings on defaults
+    if e == None: e = 0.; print('e not passed, setting to 0... (circular)')
+    if omega == None:
+        omega = 0. # only warn if orbit is non-circular
+        if e != 0.: print('omega not passed, setting to 0... (pericenter at ascending node)')
+    if inc == None: inc = 0.; print('inc not passed, setting to 0... (in xy plane)')
+    if Omega == None:
+        Omega = 0. # only warn if orbit is inclined
+        if inc != 0.: print('Omega not passed, setting to 0... (ascending node along x axis)')
+    if f != None and M != None: raise ValueError('Can only pass EITHER the true (f) or the mean (M) anomaly')
+    if f == None and M == None:
+        f = 0.;
+        if e != 0.:
+            print('Neither f nor M passed, setting initial angle from pericenter to 0...')
+        else:
+            print('Neither f nor M passed, setting particle on x axis...')
+    if mass == None: mass = 0.; print('mass not passed, setting to 0...')
+    
+    if not(0. <= e < 1.): raise ValueError('e must be in range [0,1)') # not sure if these equations work for parabolic/hyperbolic obits
+    if not(0. <= inc <= numpy.pi): raise ValueError('inc must be in range [0,pi]')
+    
+    if f == None: # need to calculate f
+        E = get_E(e,M)
+        f = mod2pi(2.*numpy.arctan(numpy.sqrt((1. + e)/(1. - e))*numpy.tan(0.5*E)))
+    
+    r = a*(1-e**2)/(1 + e*numpy.cos(f))
+    
+    # Murray & Dermott Eq. 2.122
+    _x  = star.x + r*(numpy.cos(Omega)*numpy.cos(omega+f) - numpy.sin(Omega)*numpy.sin(omega+f)*numpy.cos(inc))
+    _y  = star.y + r*(numpy.sin(Omega)*numpy.cos(omega+f) + numpy.cos(Omega)*numpy.sin(omega+f)*numpy.cos(inc))
+    _z  = star.z + r*numpy.sin(omega+f)*numpy.sin(inc)
+    
+    n = numpy.sqrt(get_G()*(star.m+mass)/(a**3))
+    
+    # Murray & Dermott Eq. 2.36 after applying the 3 rotation matrices from Sec. 2.8 to the velocities in the orbital plane
+    _vx = star.vx + (n*a/numpy.sqrt(1-e*e))*((e+numpy.cos(f))*(-numpy.cos(inc)*numpy.cos(omega)*numpy.sin(Omega) - numpy.cos(Omega)*numpy.sin(omega)) - numpy.sin(f)*(numpy.cos(omega)*numpy.cos(Omega) - numpy.cos(inc)*numpy.sin(omega)*numpy.sin(Omega)))
+    _vy = star.vy + (n*a/numpy.sqrt(1-e*e))*((e+numpy.cos(f))*(numpy.cos(inc)*numpy.cos(omega)*numpy.cos(Omega) - numpy.sin(omega)*numpy.sin(Omega)) - numpy.sin(f)*(numpy.cos(omega)*numpy.sin(Omega) + numpy.cos(inc)*numpy.cos(Omega)*numpy.sin(omega)))
+    _vz = star.vz + (n*a/numpy.sqrt(1-e*e))*((e+numpy.cos(f))*numpy.cos(omega)*numpy.sin(inc) - numpy.sin(f)*numpy.sin(inc)*numpy.sin(omega))
+    
+    return Particle(m  = mass, x = _x, y = _y, z = _z, vx = _vx, vy = _vy, vz = _vz)
+
+def p2orbit(p, star):
+    '''Takes 2 Particle objects (one for object and one for central body,
+        which hold Cartesian elements) and returns a Orbit object
+        with all the orbital elements'''
+    
+    if star.m <= 1.e-30: raise ValueError('Star has no mass.')
```

Comment:
I agree with Hanno, and I think we should name the magic number.  Something like

    SMALL = 1.e-300

at the top of the file, and then

    if star.m <= SMALL

etc.